src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs

```diff
@@ -208,12 +209,24 @@ private async Task InnerOnConnectionAsync(ConnectionContext context)
                     await sslStream.DisposeAsync();
                     return;
                 }
-                catch (Exception ex) when (ex is IOException || ex is AuthenticationException)
+                catch (IOException ex)
                 {
                     _logger?.LogDebug(1, ex, CoreStrings.AuthenticationFailed);
                     await sslStream.DisposeAsync();
                     return;
                 }
+                catch (AuthenticationException ex)
+                {
+                    _logger?.LogDebug(1, ex, CoreStrings.AuthenticationFailed);
+                    if (_serverCertificate != null &&
+                        CertificateManager.IsHttpsDevelopmentCertificate(_serverCertificate) &&
+                        !CertificateManager.CheckDeveloperCertificateKey(_serverCertificate))
+                    {
+                        _logger?.LogError(2, CoreStrings.BadDeveloperCertificateState);
```

Comment:
Add the exception to this log and put the other log into an else clause? No point in logging twice for the same event.

Reply:
Fair enough